docs/docs/aztec/concepts/accounts/index.md

```diff
@@ -72,87 +63,99 @@ def entryPoint(payload):
+However, this is not required when sitting on the receiving end. The user address can be deterministically derived from their encryption public key and the account contract they intend to deploy. So that the funds can be sent to an account that hasn't been deployed yet.
 
-As an example, we can think of a lottery contract, where at some point a prize needs to be paid out to its winners. This `pay` action does not require authentication and does not need to be executed by any user in particular, so anyone could submit a transaction that defines the lottery contract itself as `origin` and `pay` as entrypoint function. For an example of this behavior see our [non_contract_account test](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/end-to-end/src/e2e_non_contract_account.test.ts) and the [SignerLess wallet](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/aztec.js/src/wallet/signerless_wallet.ts) implementation.
-Notice that the Signerless wallet doesn't invoke an entrypoint function of an account contract but instead invokes the target contract function directly.
+Users will need to pay transaction fees in order to deploy their account contract. This can be done by sending fee juice to their account contract address (which can be derived deterministically, as mentioned above), so that the account has funds to pay for its own deployment. Alternatively, the fee can be paid for by another account, using [fee abstraction](#fee-abstraction).
 
-:::info
-Entrypoints for the following cases:
+## What is account address
 
-- if no contract entrypoint is used `msg_sender` is set to `Field.max`.
-- in a private to public entrypoint, `msg_sender` is the contract making the private to public call
-- when calling the entrypoint on an account contract, `msg_sender` is set to the account contract address
-  :::
+Address is derived from the [address keys](keys.md#address-keys). While the AddressPublicKey is an elliptic curve point of the form (x,y) on the [Grumpkin elliptic curve](https://github.com/AztecProtocol/aztec-connect/blob/9374aae687ec5ea01adeb651e7b9ab0d69a1b33b/markdown/specs/aztec-connect/src/primitives.md), the address is its x coordinate. The corresponding y coordinate can be derived if needed. For x to be legit as an address there should exist a corresponding y that satisfies the curve equation: any field element cannot work as an address.
 
-### Account initialization
+### Complete address
 
-The protocol requires that every account is a contract for the purposes of sending a transaction. This means that a user needs to deploy their account contract as their first action when they want to interact with the network.
+Because of the contract address derivation scheme you can check that a given set of public [keys](keys.md) corresponds to a given address just by trying to recompute it.
```

Comment:
```suggestion
Because of the contract address derivation scheme, you can check that a given set of public [keys](keys.md) corresponds to a given address by trying to recompute it.
```